### PR TITLE
feat: add Treemap viz

### PR DESCRIPTION
**Pull request by plagoa**

### CODE REVIEW

File: packages/viz/src/Treemap/Treemap.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      height,
+      tooltip,
+      onOptionChange,
+      ...others
+    } = props;
+
+    const { colors } = useTheme();
+    const chartTooltip = useTooltip({
+      ...tooltip,
+      type: "single",
```
Comment (MEsteves22) [inline packages/viz/src/Treemap/Treemap.tsx:58]: The `type` defined by the user (inside `tooltip`) will be overridden. If there's no `multiple` tooltip type for the treetop, we should remove it.

**Review by MEsteves22 — COMMENTED** [review]
LGTM👌🏻I just have the following comments:

File: packages/viz/src/Treemap/Treemap.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+import ReactECharts from "echarts-for-react/lib/core";
+import { TreemapChart } from "echarts/charts";
+import { TooltipComponent } from "echarts/components";
+import * as echarts from "echarts/core";
+
+import { HvBaseChart } from "../BaseChart";
+import { HvChartTooltipClasses, useOption, useTooltip } from "../hooks";
+import { HvChartTooltip } from "../types";
+import { HvChartCommonProps } from "../types/common";
+import { HvTreemapData } from "../types/generic";
```
Comment (zettca) [inline packages/viz/src/Treemap/Treemap.tsx:11]: This `HvTreemapData` look specific to this chart and not generic

**Review by zettca — APPROVED** [review]
LGTM 👍

File: packages/viz/src/Treemap/Treemap.tsx (hunk 2 above)
Comment (plagoa) [reply packages/viz/src/Treemap/Treemap.tsx:11]: You're right, I'll update that.

**Review by MEsteves22 — APPROVED** [review]

**Pull request merged**